ui/src/views/Quickstart/index.jsx

```diff
@@ -210,29 +209,30 @@ export default class QuickStart extends Component {
   renderEditor() {
     const newYaml = safeDump({
       ...initialYaml,
-      access: this.state.access,
-      tasks: [
-        {
-          ...initialYaml.tasks[0],
-          ...{
-            metadata: {
-              ...initialYaml.tasks[0].metadata,
-              name: this.state.taskName,
-              description: this.state.taskDescription,
-            },
-            extra: {
-              github: {
-                events: [...this.state.events].sort(),
+      policy: {
+        pullRequests: this.state.access,
+      },
+      tasks: {
+        $match: {
+          [`tasks_for == "github-pull-request" && event["action"] in ${[
```

Comment:
here pull request is hard-coded. We want the group of checkmarks `This task should run on` to work: 
* if you tick `Push` (for example) checkmark, this line should have `tasks_for == "push"` in here.
* if you untick all those checkmarks, it should be an empty string
* `event["action"]` is relevant for PRs only. If no pull-request -related checkmark is ticked, this condition should not be there

Please refer to this section of documentation: https://github.com/taskcluster/taskcluster/blob/master/services/github/docs/taskcluster-yml-v1.md#github-events

Reply:
What if multiple checkboxes are checked, then how it needs to be handled?
`tasks_for == "github-pull-request" || tasks_for == "github-push"` 
Something like this?

Reply:
Yes! The only thing, there are several checkboxes that are all about PRs, so `tasks_for` is `github-pull-request`, but each checkbox has it's own action. I would put PR-related expression in parenthesis, like `(tasks_for ... && event["action"] ...) || tasks_for ...` - something like that.
 
as far json-e goes, you can refer to its boolean expressions docs here https://taskcluster.github.io/json-e/#Language/boolean-operations (and that website contains much more docs)